WeeChat fails to build on OS X

## Issue description
WeeChat fails to build on OS X with the below error

```
[ 43%] Building C object src/plugins/irc/CMakeFiles/irc.dir/irc-server.o
[ 44%] Building C object src/plugins/irc/CMakeFiles/irc.dir/irc-upgrade.o
[ 44%] Linking C shared module irc.so
Undefined symbols for architecture x86_64:
  "_res_9_init", referenced from:
      _irc_server_xfer_send_ready_cb in irc-server.o
ld: symbol(s) not found for architecture x86_64
clang-3.7: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/plugins/irc/CMakeFiles/irc.dir/build.make:694: src/plugins/irc/irc.so] Error 1
make[1]: *** [CMakeFiles/Makefile2:703: src/plugins/irc/CMakeFiles/irc.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
builder for ‘/nix/store/3nirbm7p4c8sdfx2k0171hkbrddhi9r0-weechat-1.6.drv’ failed with exit code 2
error: build of ‘/nix/store/3nirbm7p4c8sdfx2k0171hkbrddhi9r0-weechat-1.6.drv’ failed
```


### Steps to reproduce
```
nix-env -i weechat
```


## Technical details

* System: macOS Sierra
* Nix version: 1.11.4
* Nixpkgs version: 17.03pre93141.e3deccc

I didn't know how to read the message, I was searching for `Undefined symbols for architecture x86_64`.

Searching for `res_9_init` I think it comes from `libresolv` on Darwin searching around.

I can see `libresolv` listed in `pkgs/os-specific/darwin/apple-source-releases/default.nix`, `pkgs/os-specific/darwin/apple-source-releases/libresolve/default.nix` and `pkgs/os-specific/darwin/apple-source-releases/Libsytem/default.nix` .

Only just started using nixpkgs so not sure how to pull it in to the Weechat derivation and test etc.

--- a/pkgs/applications/networking/irc/weechat/default.nix
+++ b/pkgs/applications/networking/irc/weechat/default.nix
@@ -1,6 +1,6 @@
 { stdenv, fetchurl, ncurses, openssl, aspell, gnutls
 , zlib, curl , pkgconfig, libgcrypt
-, cmake, makeWrapper, libobjc, libiconv
+, cmake, makeWrapper, libobjc, libresolv, libiconv
 , asciidoctor # manpages
 , guileSupport ? true, guile
 , luaSupport ? true, lua5
@@ -50,7 +50,7 @@ stdenv.mkDerivation rec {
       cmake
       asciidoctor
       ]
-    ++ optionals stdenv.isDarwin [ pync libobjc ]
+    ++ optionals stdenv.isDarwin [ pync libobjc libresolv ]
     ++ optional  guileSupport    guile
     ++ optional  luaSupport      lua5
     ++ optional  perlSupport     perl
@@ -60,7 +60,7 @@ stdenv.mkDerivation rec {
 
   NIX_CFLAGS_COMPILE = "-I${python}/include/${python.libPrefix}"
     # Fix '_res_9_init: undefined symbol' error
-    + (stdenv.lib.optionalString stdenv.isDarwin "-DBIND_8_COMPAT=1");
+    + (stdenv.lib.optionalString stdenv.isDarwin "-DBIND_8_COMPAT=1 -lresolv");
 
   postInstall = with stdenv.lib; ''
     NIX_PYTHONPATH="$out/lib/${python.libPrefix}/site-packages"

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -14971,6 +14971,7 @@ in
 
   weechat = callPackage ../applications/networking/irc/weechat {
     inherit (darwin) libobjc;
+    inherit (darwin) libresolv;
   };
 
   westonLite = callPackage ../applications/window-managers/weston {

